src/main/java/Parser.java

```diff
@@ -0,0 +1,307 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Scanner;
+
+public class Parser {
+    private Ui ui;
+
+    public Parser(Ui ui) {
+        this.ui = ui;
+    }
+
+    /**
+     * Processes inputs from the user.
+     * @param tasks TaskList containing all tasks.
+     * @param filePath for path of file.
+     */
+    public void parser(TaskList tasks, String filePath) {
+        Scanner scanner = new Scanner(System.in);
+        Path path = Paths.get(filePath);
+
+        while (scanner.hasNextLine()) {
+            String line = scanner.nextLine();
+            if (line.equals("bye")) {
+                ui.bye();
+                break;
+            } else if (line.equals("list")) {
+                ui.list(tasks);
+            } else {
+                String[] words = line.split("\\s+");
+                if (words[0].equals("done")) {
+                    try {
+                        int index = completeDelete(line, tasks.size());
+                        tasks.get(index).setDone();
+                        Storage.fileUpdate(tasks, path);
+                        ui.done(tasks, index);
+                    } catch (DukeException e) {
+                        System.out.println(e.getMessage());
+                    }
+                } else if (words[0].equals("delete")) {
+                    try {
+                        int index = completeDelete(line, tasks.size());
+                        ui.remove(tasks, index);
+                        tasks.delete(index);
+                        Storage.fileUpdate(tasks, path);
+                    } catch (DukeException e) {
+                        System.out.println(e.getMessage());
+                    }
+                } else if (words[0].equals("find")) {
+                    if (words.length > 1) {
+                        String keyword = line.substring(5);
+                        TaskList matches = matchFinder(tasks, keyword);
+                        ui.foundMatches(matches);
+                    }
+                } else {
+                    try {
+                        Task task = taskClassify(line);
+                        tasks.add(task);
+                        Storage.fileUpdate(tasks, path);
+                        ui.add(task, tasks.size());
+                    } catch (DukeException e) {
+                        System.out.println(e.getMessage());
```

Comment:
Sections like these are quite deeply nested, maybe you might want to refactor this? The other methods in this class also have this issue. See [the textbook page](https://nus-cs2103-ay2021s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-deep-nesting) for the section on deep nesting.